src/classes/USER_UserService.cls

```diff
@@ -71,15 +71,33 @@ public with sharing class USER_UserService {
     /*********************************************************************************************************
     * @description Returns user Customize Application permission
     * @param profileId User profile Id
-    * @return Boolean True if the user has Customize Application permission
+    * @return Boolean True if the user has Customize Application permission at profile level or 
+    * an assigned custom permission set
     **********************************************************************************************************/
-    public static Boolean hasCustomizeApplicationPermission(Id profileId) {
+    public static Boolean isCustomizeApplicationUser() {
+        Id profileId = UserInfo.getProfileId();
+        Id userId = UserInfo.getUserId();
+
         Profile userProfile = [
             SELECT Id, PermissionsCustomizeApplication
             FROM Profile
             WHERE Id = :profileId];
 
-        return userProfile != null ? userProfile.PermissionsCustomizeApplication : false;
+        if (userProfile.PermissionsCustomizeApplication == true) {
```

Comment:
@RadSFDeveloper The above query against Profile is not necessary because the query below against PersmissionSetAssignment actually includes both Profiles and Permission Sets.